content/usgs-packages/sbtools_Discovery.Rmd

```diff
@@ -0,0 +1,224 @@
+
+```{r}
+### SPATIAL QUERY EXAMPLES NEED SOME TLC
+
+appalachia <- data.frame(
+  lat = c(34.576900, 36.114974, 37.374456, 35.919619, 39.206481),
+  long = c(-84.771119, -83.393990, -81.256731, -81.492395, -78.417345))
+
+conus <- data.frame(
+  lat = c(49.078148, 47.575022, 32.914614, 25.000481),
+  long = c(-124.722111, -67.996898, -118.270335, -80.125804))
+
+# verifying where points are supposed to be
+maps::map('usa')
+points(conus$long, conus$lat, col="red", pch=20)
+points(appalachia$long, appalachia$lat, col="green", pch=20)
+
+# query by bounding box
+bbox_sp_obj <- sp::SpatialPoints(appalachia)
+# query_sb_spatial(bbox=bbox_sp_obj)
+
+# query by latitude and longitude vectors
+query_sb_spatial(long = appalachia$long, lat = appalachia$lat)
+query_sb_spatial(long = conus$long, lat = conus$lat)
+
+# query by WKT polygon
+wkt_coord_str <- paste(conus$long, conus$lat, sep=" ", collapse = ",")
+wkt_str <- sprintf("POLYGON((%s))", wkt_coord_str)
+# query_sb_spatial(bb_wkt = wkt_str)
+```
+
+### Using `query_sb_date`
+
+`query_sb_date` returns ScienceBase items that fall within a certain time range. There are multiple timestamps applied to items, so you will need to specify which one to match the range. The default queries are to look for items last updated between 1970-01-01 and today's date. See `?query_sb_date` for more examples of which timestamps are available.
+
+```{r}
+# find data worked on in the last week
+today <- Sys.time()
+oneweekago <- today - (7*24*3600) # days * hrs/day * secs/hr
+recent_data <- query_sb_date(start = today, end = oneweekago)
+head(recent_data, 2)
+
+# find data that's been created over the last year
+oneyearago <- today - (365*24*3600) # days * hrs/day * secs/hr
+recent_data <- query_sb_date(start = today, end = oneyearago, date_type = "dateCreated")
+head(recent_data, 2)
+```
+
+### Using `query_sb_datatype`
+
+`query_sb_datatype` is used to search ScienceBase by the type of data an item is listed as. Run `sb_datatypes()` to get a list of 50 available data types.
+
+```{r}
+# get ScienceBase news items
+sbnews <- query_sb_datatype("News")
+head(sbnews, 2)
+
+# find shapefiles
+shps <- query_sb_datatype("Shapefile")
+head(shps, 2)
+
+# find raster data
+sbraster <- query_sb_datatype("Raster")
+head(sbraster, 2)
+```
+
+## Best of both methods
+
+Although you can query from R, sometimes it's useful to look an item on the web interface. You can use the `query_sb_*` functions and then follow that URL to view items on the web. This is especially handy for viewing maps and metadata, or to check or repair a ScienceBase item if any of the `sbtools`-based commands have failed.
+
+```{r}
+sbmaps <- query_sb_datatype("Static Map Image", limit=3)
+oneitem <- sbmaps[[1]]
+
+# get and open URL from single sbitem
+url_oneitem <- oneitem[['link']][['url']]
+browseURL(url_oneitem)
+
+# get and open URLs from many sbitems
+lapply(sbmaps, function(sbitem) {
+  url <- sbitem[['link']][['url']]
+  browseURL(url)
+  return(url)
```

Comment:
Was thinking that it would be nice to just print out the URLs that were opened. I don't know why, but I generally don't like functions that don't return anything.